src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_TryCatch.cs

```diff
@@ -116,5 +120,638 @@ void M(int i)
+  ExceptionDeclarationOrExpression: IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration) (Syntax: '(System.IO. ... xception e)')
+      Variables: Local_1: System.IO.IOException e
+      Initializer: null
+  Filter: IBinaryOperatorExpression (BinaryOperatorKind.NotEquals) (OperationKind.BinaryOperatorExpression, Type: System.Boolean) (Syntax: 'e.Message != null')
+      Left: IPropertyReferenceExpression: System.String System.Exception.Message { get; } (OperationKind.PropertyReferenceExpression, Type: System.String) (Syntax: 'e.Message')
+          Instance Receiver: ILocalReferenceExpression: e (OperationKind.LocalReferenceExpression, Type: System.IO.IOException) (Syntax: 'e')
+      Right: IConversionExpression (Implicit, TryCast: False, Unchecked) (OperationKind.ConversionExpression, Type: System.String, Constant: null) (Syntax: 'null')
+          Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: False, IsReference: True, IsUserDefined: False) (MethodSymbol: null)
+          Operand: ILiteralExpression (OperationKind.LiteralExpression, Type: null, Constant: null) (Syntax: 'null')
+  Handler: IBlockStatement (0 statements) (OperationKind.BlockStatement) (Syntax: '{ ... }')
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<CatchClauseSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void TryCatch_GetOperationForCatchDeclaration()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        try
+        {
+        }
+        catch /*<bind>*/(System.IO.IOException e)/*</bind>*/ when (e.Message != null)
+        {
+        }
+    }
+}
+";
+            string expectedOperationTree = @"
+IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration) (Syntax: '(System.IO. ... xception e)')
+  Variables: Local_1: System.IO.IOException e
+  Initializer: null
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<CatchDeclarationSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/22299")]
+        public void TryCatch_GetOperationForCatchFilterClause()
+        {
+            string source = @"
+using System;
+
+class C
+{
+    static void M(string s)
+    {
+        try
+        {
+        }
+        catch (Exception) /*<bind>*/when (s != null)/*</bind>*/
+        {
+        }
+    }
+}
+";
+            string expectedOperationTree = @"
+Filter: IBinaryOperatorExpression (BinaryOperatorKind.NotEquals) (OperationKind.BinaryOperatorExpression, Type: System.Boolean) (Syntax: 's != null')
```

Comment:
I would expect to get nothing here, there is no node associated with CatchFilterClauseSyntax.

Reply:
Currently, we do return null, so I had skipped this test with [https://github.com/dotnet/roslyn/issues/22299](https://github.com/dotnet/roslyn/issues/22299). I will close that issue and update this test to verify null.

---
In reply to: [141130773](https://github.com/dotnet/roslyn/pull/22210#discussion_r141130773) [](ancestors = 141130773)